File: DSA_with_C++/Linked_LIST/insert_at_tail.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def insert_at_tail(head, v):
    new_node = Node(v)
    if head is None:
        return new_node
    temp = head
    while temp.next is not None:
        temp = temp.next
    temp.next = new_node
    return head

def print_linked_list(head):
    print("Your linked list:", end=" ")
    temp = head
    while temp is not None:
        print(temp.val, end=" ")
        temp = temp.next
    print()

def main():
    head = None
    while True:
        print("Option 1: Insert at tail")
        print("Option 2: Print linked list")
        print("Option 3: Terminate")
        op = int(input("Choose an option: "))
        if op == 1:
            v = int(input("Please enter value: "))
            head = insert_at_tail(head, v)
        elif op == 2:
            print_linked_list(head)
        elif op == 3:
            break
        else:
            print("Invalid option.")

if __name__ == "__main__":
    main()

File: DSA_with_C++/Linked_LIST/insert_at_tail.cpp
#include<bits/stdc++.h>
using namespace std;

class Node{
    public:
    int val;
    Node* next;

    Node(int val){
        this->val=val;
        this->next=NULL;
    }
};
void insert_at_tail(Node *&head,int v)
{
Node *newNode = new Node(v);
if(head==NULL){
    head = newNode;
    return;
}
Node * temp = head;
while(temp->next !=NULL){
    temp= temp->next; 
   
     
}
 //temp now last node 
 temp->next= newNode;
}
void print_linked_list(Node *head){
    cout<<"Your linked list: ";
    Node *temp =head;
    while(temp !=NULL){
        cout<<temp->val<<" ";
        temp=temp->next;
    }
    cout<<endl;
}
int main()
{
    Node* head = NULL;
   while(true){
    cout<<"Option 1: insert at tail"<<endl;
    cout<<"Option 2: print linked list "<<endl;
    cout<<"Option 3: Terminate "<<endl;
    int op;
    cin>>op;
    if(op==1){
        cout<<"please enter value: "<<endl;
        int v;
        cin>>v;
        insert_at_tail(head,v);
    }
    else if(op==2){
        print_linked_list(head);
    }
    else if(op==3){
        break;
    }
   }
    return 0;
}